.github/workflows/android-build.yml

```diff
@@ -76,6 +76,10 @@ jobs:
           path: example/android/app/build
           key: ${{ runner.os }}-build-${{ env.cache-name }}-example-${{ hashFiles('**/react-native/package.json') }}
   
+      - name: Build Android app
+        working-directory: example/android
+        run: ./gradlew assembleDebug
```

Comment:
If this takes 20 minutes, I would build in a separate job, publish the APK as artifact and then download the artifact separately, or use Gradle cache to make sure the whole building running FROM-CACHE.